### DISCUSSION TITLE
为什么需要稀疏向量的HNSW索引？Why do we need HNSW index for sparse vector?

### DISCUSSION THREAD

**Discussion by dtad5** [milvus-io/milvus#34372] [Q&A and General discussion] [answered]
最近发布的milvus更新增加了对稀疏向量-hnsw算法的支持，在倒排类索引适合关键词搜索的情况下，请问一下为什么或者说什么场景有使用稀疏hnsw类图算法的需求？
Sparse-hnsw algorithm is added in the latest milvus update. Given that inverted index is suitable for search with terms, on what condition do we need graph algorithms like hnsw for sparse vector fileds?

**Comment by yhmo** [comment]
Sparse-hnsw? I don't think milvus has this index type.

Officially, only two types are optional: SPARSE_INVERTED_INDEX and SPARSE_WAND
https://milvus.io/docs/index-vector-fields.md?tab=sparse

**Reply by dtad5** [reply]
Thank you for the reply! Actually I just read about the second new feature in release v2.4.5: [https://milvus.io/docs/release_notes.md](https://milvus.io/docs/release_notes.md) Maybe this is only for beta version, or did I misunderstand something?

**Reply by yhmo** [reply, marked as answer]
It is a mistake of the release note.
The release note mentioned: Supported building the HNSW index for sparse vectors ([#33653](https://github.com/milvus-io/milvus/pull/33653), [#33662](https://github.com/milvus-io/milvus/pull/33662))

The description of the pr #33662 is: "support sparse cardinal hnsw index"
The cardinal index is derived from hnsw, and the cardinal index is only available in Zilliz cloud. So, the "sparse cardinal hnsw index" is only for Zilliz cloud.
Opensource Milvus doesn't have cardinal index.

**Reply by xiaofan-luan** [reply]
> Thank you for the reply! Actually I just read about the second new feature in release v2.4.5: https://milvus.io/docs/release_notes.md Maybe this is only for beta version, or did I misunderstand something?

Sorry for the misunderstanding.
Graph based index can work together with Sparse embedding, and turns out to be much faster.
But this is for now Zilliz cloud feature only

**Reply by xiaofan-luan** [reply]
for opensource, WAND index might be a good solution and it's faster than es
